internal/handlemap/handle_map.go

```diff
@@ -156,6 +160,30 @@ func (handle *Handle) Cleanup() {
 	}
 }
 
+// **********************Dcache Related Methods Start******************************
+func (handle *Handle) SetFsAzure() {
+	handle.Flags.Set(HandleFSAzure)
```

Comment:
add these asserts before setting

// Must be set once and only once.
common.Assert(!handle.IsFsAzure())
common.Assert(!handle.IsFsDcache())

add the same in SetFsDcache() and SetFsDefault() too.